### PR TITLE
feat(PLM-1415): Support validating multiple row data by dict-like Sp…

### PR DESCRIPTION
**Pull request by kilikkuo**
…ec natuarally

[Why]
DSV now validates Spec and treats the input value as a dictionary. We need to make another spec containing this list data for validation.

[How]
1. Provide multirow option for decorator dsv & function validate_data_spec
2. Add test & revise README

[PLM-1415]

Reason
--------------

Changes
--------------

Test Scope
--------------

Checks
--------------
- [x] Unit tests are included or not applicable.

[PLM-1415]: https://hardcoretech.atlassian.net/browse/PLM-1415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kilikkuo** [comment]
oops, 太久沒上這個 repo 的 PR, initial pre-commit 竟然失敗惹 XD

**Comment by cmj0121** [comment]
> oops, 太久沒上這個 repo 的 PR, initial pre-commit 竟然失敗惹 XD

need to change the pre-commit repo from gitlab.com to github.com

**Comment by kilikkuo** [comment]
> > oops, 太久沒上這個 repo 的 PR, initial pre-commit 竟然失敗惹 XD
> 
> need to change the pre-commit repo from gitlab.com to github.com

THANKS :)

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -4,12 +4,14 @@ on: [pull_request]
...
+    runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 1024
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.6
```
Comment (kilikkuo) [inline .github/workflows/lint.yml:14]: 為了避免使用 fms 還不支援的 python future, 鎖在 3.6 先

File: test/test_decorator.py [hunk 2]
```diff
@@ -119,6 +120,42 @@ def decorated_func(self, request, *_args, **_kwargs):
...
+        class _ViewSingleRowSpec:
+            test_a = Checker([STR])
+
+        class _View(View):
+            @dsv(_ViewSingleRowSpec)
+            def decorated_func(self, request, *_args, **_kwargs):
+                pass
+
+        view = _View(request=fake_request)
+        view.decorated_func(fake_request, **kwargs)
```
Comment (kilikkuo) [inline test/test_decorator.py:138]: Test for backward-compatibility

File: test/test_decorator.py [hunk 3]
```diff
@@ -119,6 +120,42 @@ def decorated_func(self, request, *_args, **_kwargs):
...
+
+        class _View(View):
+            @dsv(_ViewSingleRowSpec, multirow=True)
+            def decorated_func(self, request, *_args, **_kwargs):
+                pass
+
+        view = _View(request=fake_request)
+
+        with patch('data_spec_validator.decorator.decorators._is_data_type_list', return_value=False):
+            view.decorated_func(fake_request, **kwargs)
```
Comment (kilikkuo) [inline test/test_decorator.py:157]: Test to ensure `multirow` option works

**Review by cmj0121 — APPROVED** [review]
LGTM thanks

**Comment by kilikkuo** [comment]
Thanks :)

**Pull request merged**